persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java

```diff
@@ -489,7 +489,7 @@ private ClientState cloneWithKeyspace(ClientState original, String keyspace) {
           original.isInternal
               ? ClientState.forInternalCalls()
               : ClientState.forExternalCalls(original.getRemoteAddress());
-      clone.login(clone.getUser());
+      clone.login(original.getUser());
```

Comment:
what's the reason behind this change? Does the `clone` not contain the same user? 
If so, maybe the method responsible for cloning should be fixed as well?

Reply:
This is the method responsible for the clone :). The line above initializes `clone` with a default ClientState, it's not until we call `login()` that a user is added so what was happening is we were calling `login()` with a null user.